src/catchup/VerifyLedgerChainWork.cpp

```diff
@@ -178,10 +202,15 @@ VerifyLedgerChainWork::verifyHistoryOfSingleCheckpoint()
         if (curr.header.ledgerVersion >
```

Comment:
Should we also set `mLocalFailure` when we catch an xdr exception? Or does it not matter because we won't be able to see if we can set `mHasTrustedHash` to true anyway?

Reply:
Yeah, my thinking was that we won't be able to verify that the hash is trusted, and given that fatal failure causes us to _never_ retry catchup (which is serious if done erroneously), we should allow catchup to retry in this case.

Reply:
Hmm, doesn't this mean validators that are sufficiently old that that they can't parse the current network header will retry forever? (I had the same concern here).

(I don't mind leaving this as a followup if so, but I think it's at least a reasonably-plausible case we currently don't handle..)

Reply:
ok